File: algorithms/solutions/week30/taurandat/mini-max-sum.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
  long long n, _sum, _min, _max;

  cin >> n;
  _sum = _min = _max = n;
  for(int i = 0; i < 4; i++) {
    cin >> n;
    _sum += n;
    _min = min(_min, n);
    _max = max(_max, n);
  }

  cout << _sum - _max << " " << _sum - _min << endl;

  return 0;
}

File: algorithms/solutions/week30/manhdv/mini_max_sum.rb
# https://www.hackerrank.com/challenges/mini-max-sum/problem

#!/bin/ruby

require 'json'
require 'stringio'

# Complete the miniMaxSum function below.
def miniMaxSum(arr)
  min = 10**9 + 1
  max = 0
  sum = 0

  arr.each do |num|
    sum += num
    max = num if num > max
    min = num if num < min
  end

  print sum - max
  print ' '
  print sum - min
end

arr = gets.rstrip.split(' ').map(&:to_i)

miniMaxSum arr
